src/components/ha-menu.ts

```diff
@@ -1,9 +1,32 @@
 import { MdMenu } from "@material/web/menu/menu";
+import {
+  CloseReason,
+  KeydownCloseKey,
+} from "@material/web/menu/internal/controllers/shared";
 import { css } from "lit";
 import { customElement } from "lit/decorators";
 
 @customElement("ha-menu")
 export class HaMenu extends MdMenu {
+  connectedCallback(): void {
+    super.connectedCallback();
+    this.addEventListener("close-menu", this._handleCloseMenu);
+  }
+
+  disconnectedCallback(): void {
+    super.disconnectedCallback();
+    this.removeEventListener("close-menu", this._handleCloseMenu);
```

Comment:
I guess It's still ok to attach on `connected`? I was slightly concerned that would create duplicate listeners but it does not appear to do so.

Reply:
> I guess It's still ok to attach on `connected`? I was slightly concerned that would create duplicate listeners but it does not appear to do so.

It would duplicate if the handler were anonymous, but since it's named here, it's fine.